### DISCUSSION TITLE
inserting vectors into partitions gives random results

### DISCUSSION THREAD

**Discussion by YangIsNotAvailable** [milvus-io/milvus#7018] [Q&A and General discussion]
Hi,

The following standalone script create a test collection and dump some fake data into a number of partitions. Finally I clean up after myself by drop the test collection.  But for some reason all not data are inserted and which ones are not inserted is random. Would love to know what's happening. Thanks!

```
from milvus import Milvus, IndexType, MetricType, Status
from pprint import pprint
import random

random.seed(2021)

client = Milvus(host='localhost', port='19530')
```

## create an empty test collection
```
test_collection = "test_050821"
dimension = 1024

test_param = {
	'collection_name': test_collection,
	'dimension': dimension,
	'index_file_size': 2048,
	'metric_type': MetricType.L2
}

print(f">>> Creating collection {test_collection}")
client.create_collection(test_param)
print(f">>> Collection {test_collection} created.")
```

## make some fake data
```
fake_data = [(i, [random.random() for _ in range(dimension)]) for i in range(10_000)]
```

## make 10 partitions
```
num_partitions = 10

for i in range(num_partitions):
    client.create_partition(
        collection_name=test_collection,
        partition_tag=str(i),
    )

status, info = client.get_collection_stats(test_collection)
if status.OK():
    print(f">>> Initial status of collection {test_collection}:")
    pprint(info)
else:
    print(status)
```

## divide data into 10 partitions by mod 10
```
groups = dict()

for i in range(num_partitions):
    groups[str(i)] = {
        "vectors": [vector for j, vector in fake_data if j % num_partitions == i],
        "ids": [j for j, vector in fake_data if j % num_partitions == i],
    }
```

## insert data into milvus
```
for i, data in groups.items():
    client.insert(
        collection_name=test_collection,
        partition_tag=i,
        records=data['vectors'], 
        ids=data['ids']
    )

status, info = client.get_collection_stats(test_collection)
if status.OK():
    print(f">>> After insertion status of collection {test_collection}:")
    pprint(info)
else:
    print(status)
```

## delete the test_collection
```
client.drop_collection(test_collection)
print(f">>> Collection {test_collection} deleted.")
```

The expected result is every partition gets 1k rows. But the actual result is that some partitions get 1k rows and some get 0 rows, and the ones with 1k rows always come before the ones with 0 rows as far as I've tested.

**Comment by yhmo** [comment]
Your milvus version is 1.x. 
In 1.x, after you invoked insert() method, data firstly inserted into a buffer, but the buffer is not visible for db engine. Only after the buffer be flushed/persisted into disk, the data can be searchable.
There is an auto-flush action in background, flush by 1 second interval(default). So, when you continually invoked insert(), some data may be flushed by auto-flush thread, some other not. 
To ensure all the data is flushed into disk, you can manually call flush(). Your client code can be like this:

```
for i, data in groups.items():
    client.insert(
        collection_name=test_collection,
        partition_tag=i,
        records=data['vectors'], 
        ids=data['ids']
    )

client.flush(collection_name=test_collection)
status, info = client.get_collection_stats(test_collection)
......
```
